deploy/lib/htsget-lambda-stack.ts

```diff
@@ -151,9 +166,27 @@ export class HtsgetLambdaStack extends Stack {
 
     const s3BucketPolicy = new PolicyStatement({
       actions: ["s3:List*", "s3:Get*"],
-      resources: settings.s3BucketResources ?? ["arn:aws:s3:::*"],
```

Comment:
Hm, flipping this option might affect the defaults we discussed in #258 with GA4GH? :-S

We should really decide on whether to just copy the example objects over and have a reasonable retention policy that doesn't destroy data accidentally :-S

Reply:
I just saw you introduced a `createBuckets[]` option, so seems that you went ahead and implemented bucket creation by default in this PR? But then the default settings don't give access to that newly created bucket?

Reply:
I've sort of tried to make this as reasonable as possible. It only creates a bucket if you want, and the bucket can't get destroyed accidentally.

Do you think it should be different somehow? I left the permissions manual because the `s3BucketResources` may already contain a wildcard that encompasses that bucket. Although, I guess adding it in additionally shouldn't hurt.

Reply:
Note that bucket creation is not the default, you have to explicitly ask for that option.

Reply:
The reason why I implemented it was because there is already a `lookupHostedZone` option which deals with stateful resources, and I thought bucket creation was similar.

Reply:
I think that if we go that route, we may as well copy the example data over and give perms to only that `data/` prefix?

I guess my point is either all or none:

all) The user deploys and has a server ready to go/working with example data, permissions included for that data: All works at the risk of defaults not being what user really wanted.

none) We provide options to create that bucket and no default permissions: Nothing works by default, DIY if you want it to work by default. Encourages users to understand config flags.

I guess that **all** option is generally more attractive after thinking about our interactions with GA4GH IT team? So far I feel we've defaulted to a half-none, which is tricky and yields issues :/

I hope this whole UX argument makes sense?

Reply:
Don't get me wrong, thanks a metric ton for those changes, good effort!!

Reply:
No I think that makes sense. I agree it should work out of the box, or not. In that case, I'll change it so that the permissions are added. It is already be including the data into the bucket (see the `BucketDeployment` piece of code).

In this instance, do you think it would be better to have the "all" option the default? I.e. it deploys and works out of the box? This would entail making the default public too (i.e. no authorizer deployed).

Reply:
One slight problem is that buckets are globally unique. So if the `settings.ts` hard-codes a bucket name it will only work once. It could be something like `htsget-rs-data-<random suffix>`.

Reply:
Yeah, let CDK generate that random bucket suffix.

Reply:
In that case maybe it makes sense to just have a boolean flag, in order to create one bucket and copy data? E.g. `createBucket`?

Reply:
createBucketAndCopyExampleData.... lol ... separate boolean options for those two, yeah.